src/status_im/contexts/wallet/tokens/events.cljs

```diff
@@ -0,0 +1,142 @@
+(ns status-im.contexts.wallet.tokens.events
+  (:require [re-frame.core :as rf]
+            [status-im.constants :as constants]
+            [status-im.contexts.wallet.common.validation :as wallet-validation]
+            [status-im.contexts.wallet.tokens.data :as tokens-data]
+            [status-im.contexts.wallet.tokens.effects]
+            [taoensso.timbre :as log]
+            [utils.ethereum.chain :as chain]))
+
+(rf/reg-event-fx
+ :wallet.tokens/get-token-list
+ (fn [{:keys [db]}]
+   {:db (assoc-in db [:wallet :ui :loading :token-list] true)
+    :fx [[:json-rpc/call
+          [{:method     "wallet_getTokenList"
+            :params     []
+            :on-success [:wallet.tokens/store-token-list]
+            :on-error   [:wallet.tokens/get-token-list-failed]}]]]}))
+
+(rf/reg-event-fx
```

Comment:
This event handler really needs at least one one good unit test. https://github.com/status-im/status-mobile/blob/cc6dcc3636bdb76d0b89ffa1c19841b3de267736/doc/tests/tests-overview.md#when-to-unit-test-events

Reply:
Added a good test with multiple cases!